packages/flutter/lib/src/material/navigation_destination_data.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'navigation_bar.dart';
+import 'navigation_drawer.dart';
+import 'navigation_rail.dart';
+
+/// A navigation desitination data to reuse between diffrent material navigation
+/// widgets like NavigationBar, NavigationDrawer and NavigationRail.
+///
+/// See also:
+///
+///  * [NavigationBar]
+///  * [NavigationDrawer]
+///  * [NavigationRail]
+class NavigationDestinationData {
+  /// Creates a Navigation destination data.
+  const NavigationDestinationData({
+    required this.icon,
+    required this.label,
+    Widget? selectedIcon,
+    this.tooltip,
+  })  : selectedIcon = selectedIcon ?? icon,
+        assert(icon != null);
+
+  /// The icon of the destination.
+  ///
+  /// Typically the icon is an [Icon] or an [ImageIcon] widget. If another type
+  /// of widget is provided then it should configure itself to match the current
+  /// [IconTheme] size and color.
+  ///
+  /// If [selectedIcon] is provided, this will only be displayed when the
+  /// destination is not selected.
+  final Widget icon;
```

Comment:
Because icons in navigation bar/rail destinations are not nullable. And navigationDestinationData is created to re-use the data between bar/rail/drawer